### ISSUE TITLE
Consider serving jQuery locally by default

### ISSUE THREAD

**Issue by sagamusix** [SimpleMachines/SMF#7615]
#### Description
Given that all major browsers use cache partitioning these days, loading a 30KB resource from a CDN probably provides little to no value to most SMF installations. Yet, using the Google CDN remains the default choice when installing SMF 2.1. Due to the GDPR this also poses some legal questions, but let's ignore those for now.

Another issue with the default choice is Content Security Policy: I upgraded my forums from SMF 2.0 to SMF 2.1 recently and the upgrader initially didn't work (stuck at 0%) because I have set a rather strict Content Security Policy: JavaScript from external domains was disallowed from being executed. The upgrader requires jQuery to work, but since SMF 2.1 loads it from Google CDN by default, the upgrade didn't start. I first had to manually change my server configuration to temporarily use a different Content Security Policy.

In summary, I would suggest that serving jQuery locally should be the default for new installations / upgrades from SMF 2.0. If that's not possible, I would suggest that at the very least the SMF upgrader should check if loading jQuery failed and warn the administrator - it took me a minute to figure out why the upgrade was stuck at 0%.

**Comment by Sesquipedalian** [comment]
The suggestion to always use the local copy during upgrade makes sense to me, since the upgrader doesn't present an option to choose the jQuery source before trying to load it.

I'm less convinced about changing the default for normal forum operation. Your situation with the upgrader occurred because of a configuration that is uncommon and that must be intentionally implemented, whereas during normal operation you do have the ability to choose a jQuery source that aligns with your security policy. Since using CDNs to serve larger libraries like jQuery can in fact still improve load times if the forum is hosted on a relatively low performance server, it should probably remain the default for normal operation. Of course, if anyone can provide numbers showing that there really is no performance gain anymore, I will be glad to change my opinion.

**Comment by albertlast** [comment]
because of gdpr the default behavior should be locally,
when the performance is not good enough the admin can change to an cdn.
